src/system_structure.jl

```diff
@@ -0,0 +1,851 @@
+Constructs a `Winch` object by directly providing its physical parameters.
+
+This constructor is an alternative to creating a winch from a `Settings` object,
+allowing for more modular or programmatic creation of winch components.
+
+# Arguments
+- `idx::Int16`: Unique identifier for the winch.
+- `tether_idxs::Vector{Int16}`: Vector of indices of the tethers connected to this winch.
+- `gear_ratio::SimFloat`: The gear ratio of the winch.
+- `drum_radius::SimFloat`: The radius of the winch drum [m].
+- `f_coulomb::SimFloat`: Coulomb friction force [N].
+- `c_vf::SimFloat`: Viscous friction coefficient [Ns/m].
+- `inertia_total::SimFloat`: Total inertia of the motor, gearbox, and drum [kgm²].
+
+# Keyword Arguments
+- `tether_len::SimFloat=0.0`: Initial tether length [m].
+- `tether_vel::SimFloat=0.0`: Initial tether velocity (reel-out rate) [m/s].
+- `brake::Bool=false`: If true, the winch brake is engaged.
+
+# Returns
+- `Winch`: A new `Winch` object.
+"""
+function Winch(idx, tether_idxs, gear_ratio, drum_radius, f_coulomb, c_vf, inertia_total;
+               tether_len=0.0, tether_vel=0.0, brake=false)
+    return Winch(idx, tether_idxs, tether_len, tether_vel, 0.0, 0.0, brake, zeros(KVec3),
+                 gear_ratio, drum_radius, f_coulomb, c_vf, inertia_total, zero(SimFloat))
+end
+
+"""
+    abstract type AbstractWing
+
+Abstract base type for all wing implementations.
+
+Concrete subtypes must implement rigid body dynamics and provide a reference frame
+for attached points and groups.
+"""
+abstract type AbstractWing end
+
+"""
+    mutable struct BaseWing <: AbstractWing
+
+A rigid wing body that can have multiple groups of points attached to it.
+
+The wing provides a rigid body reference frame for attached points and groups.
+Points with `type == WING` move rigidly with the wing body according to the
+wing's orientation matrix `R_b_w` and position `pos_w`.
+
+# Special Properties
+The wing's orientation can be accessed as a rotation matrix or a quaternion:
+```julia
+R_matrix = wing.R_b_w
+wing.R_b_w = R_matrix
+
+quat = wing.Q_b_w
+wing.Q_b_w = quat
+```
+
+$(TYPEDFIELDS)
+"""
+mutable struct BaseWing <: AbstractWing
+    const idx::Int16
+
+    # Structural information
+    const group_idxs::Vector{Int16}
+    const transform_idx::Int16
+    const R_b_c::Matrix{SimFloat}
+    const pos_cad::KVec3
+
+    # Differential variables in world frame, updated during simulation
+    const Q_b_w::Vector{SimFloat}
+    const ω_b::KVec3
+    const pos_w::KVec3
+    const vel_w::KVec3
+    const acc_w::KVec3
+
+    # Derived variables and parameters, updated during simulation
+    const wind_disturb::KVec3
+    drag_frac::SimFloat
+    const va_b::KVec3 # apparent wind in body frame
+    const v_wind::KVec3 # wind velocity in world frame at the wing
+    const aero_force_b::KVec3 # aerodynamic force in body frame
+    const aero_moment_b::KVec3 # aerodynamic moment in body frame
+    const tether_moment::KVec3 # tether moment in world frame
```

Comment:
The world frame is not defined in http://localhost:8000/reference_frames.html . Do not use reference frames that are undocumented.